In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_excel("acleda.xlsx")

In [3]:
df

Date  Closing Price  Trading Volume (shr)  Opening   High    Low
0   2023-05-25          10240                 63438    10220  10240  10220
1   2023-05-24          10240                145976    10240  10240  10200
2   2023-05-23          10240                134363    10280  10300  10240
3   2023-05-22          10280                189000    10300  10320  10260
4   2023-05-19          10320                109609    10300  10360  10300
..         ...            ...                   ...      ...    ...    ...
730 2020-05-29          22600                793159    22800  23600  22000
731 2020-05-28          21550                239030    20000  21550  20000
732 2020-05-27          19600                250910    17900  19660  17900
733 2020-05-26          17880                 32246    16860  17880  16520
734 2020-05-25          16500                 15888    16200  16800  16180

[735 rows x 6 columns]

# **Apply Lump Sum Strategy**
#### Lump Sum is the strategy of going all in hoping the market price goes up.
**Assumptions**

- we only have 36000000 Riels

In [4]:
total_investments = 36000000

In [5]:
shares_bought = total_investments/df['Opening'].iloc[-1]
print("Number of Shares Bought : ", shares_bought , " Shares")

Number of Shares Bought :  2222.222222222222  Shares


In [6]:
df[df['Date'] == '2021-05-25']

Date  Closing Price  Trading Volume (shr)  Opening   High    Low
497 2021-05-25          16460                 11209    16400  16480  16380

In [7]:
year1 = shares_bought * df['Opening'].iloc[497]
print(f"The value of your shares at the end of year 1: {year1} riels \n")

r1 = round(year1/total_investments - 1, 4)
print(f"If you sell now you would get : {r1} returns")

The value of your shares at the end of year 1: 36444444.44444444 riels 

If you sell now you would get : 0.0123 returns


In [8]:
df[df['Date'] == '2022-05-25']

Date  Closing Price  Trading Volume (shr)  Opening   High    Low
251 2022-05-25          11360                122773    11340  11360  11320

In [9]:
year2 = shares_bought * df['Opening'].iloc[251]
print(f"The value of your shares at the end of year 2: {year2} riels \n")

r2 = round(year2/year1 - 1, 4)
print(f"If you sell now, your return would be : {r2} returns")

The value of your shares at the end of year 2: 25200000.0 riels 

If you sell now, your return would be : -0.3085 returns


In [10]:
df[df['Date'] == '2023-05-25']

Date  Closing Price  Trading Volume (shr)  Opening   High    Low
0 2023-05-25          10240                 63438    10220  10240  10220

In [11]:
year3 = shares_bought * df['Opening'].iloc[0]
print(f"The value of your shares at the end of year 3: {year3} riels \n")

r3 = round(year3/year2 - 1, 4)
print(f"If you sell now you would get : {r3} returns")

The value of your shares at the end of year 3: 22711111.111111112 riels 

If you sell now you would get : -0.0988 returns


# **Dollar Cost Averaging (DCA) Strategy**
#### DCA is the practice of investing a fixed dollar amount on a regular basis, regardless of the share price.

**Assumptions**
- We only have 36000000 riels
- We invest 1000000 riels per month for 36 months

In [12]:
fixed_investment = 1000000

In [13]:
m1 = fixed_investment/df['Opening'].iloc[-1]
m2 = fixed_investment/df['Opening'].iloc[712]
m3 = fixed_investment/df['Opening'].iloc[690]
m4 = fixed_investment/df['Opening'].iloc[674]
m5 = fixed_investment/df['Opening'].iloc[655]
m6 = fixed_investment/df['Opening'].iloc[635]
m7 = fixed_investment/df['Opening'].iloc[617]
m8 = fixed_investment/df['Opening'].iloc[595]
m9 = fixed_investment/df['Opening'].iloc[576]
m10 = fixed_investment/df['Opening'].iloc[553]
m11 = fixed_investment/df['Opening'].iloc[534]
m12 = fixed_investment/df['Opening'].iloc[515]

In [14]:
sh1 = m1+m2+m4+m5+m6+m7+m8+m9+m10+m11+m12
print(f"You owned a total of {sh1} shares")
v1 = sh1*df['Closing Price'].iloc[515]
print(f"The value of your shares : {v1} riels")
print("You invested", fixed_investment*12, "riels")

rd1 = round(v1/(fixed_investment*12) - 1, 4)
print(f"The return for year 1 : {rd1} returns")

You owned a total of 644.4342047354885 shares
The value of your shares : 10852272.007745625 riels
You invested 12000000 riels
The return for year 1 : -0.0956 returns


In [15]:
df[df['Date'] == '2022-04-25']

Date  Closing Price  Trading Volume (shr)  Opening   High    Low
272 2022-04-25          12360              15342879    12360  12480  12300

In [16]:
m13 = fixed_investment/df['Opening'].iloc[497]
m14 = fixed_investment/df['Opening'].iloc[475]
m15 = fixed_investment/df['Opening'].iloc[454]
m16 = fixed_investment/df['Opening'].iloc[432]
m17 = fixed_investment/df['Opening'].iloc[410]
m18 = fixed_investment/df['Opening'].iloc[394]
m19 = fixed_investment/df['Opening'].iloc[375]
m20 = fixed_investment/df['Opening'].iloc[353]
m21 = fixed_investment/df['Opening'].iloc[333]
m22 = fixed_investment/df['Opening'].iloc[310]
m23 = fixed_investment/df['Opening'].iloc[291]
m24 = fixed_investment/df['Opening'].iloc[272]

In [17]:
sh2 = m13+m14+m15+m16+m17+m18+m19+m20+m21+m22+m23+m24
print(f"You owned a total of {sh2} shares")
v2 = sh2*df['Closing Price'].iloc[272]
print(f"The value of your shares : {v2} riels")
print("You invested", fixed_investment*12, "riels")

rd2 = round(v2/(fixed_investment*12) - 1, 4)
print(f"The return for year 2 : {rd2} returns")

You owned a total of 985.556598076971 shares
The value of your shares : 12181479.552231362 riels
You invested 12000000 riels
The return for year 2 : 0.0151 returns


In [18]:
df[df['Date'] == '2023-04-25']

Date  Closing Price  Trading Volume (shr)  Opening   High    Low
19 2023-04-25          11380                 79089    11500  11520  11340

In [19]:
m25 = fixed_investment/df['Opening'].iloc[251]
m26 = fixed_investment/df['Opening'].iloc[229]
m27 = fixed_investment/df['Opening'].iloc[209]
m28 = fixed_investment/df['Opening'].iloc[186]
m29 = fixed_investment/df['Opening'].iloc[164]
m30 = fixed_investment/df['Opening'].iloc[144]
m31 = fixed_investment/df['Opening'].iloc[124]
m32 = fixed_investment/df['Opening'].iloc[103]
m33 = fixed_investment/df['Opening'].iloc[81]
m34 = fixed_investment/df['Opening'].iloc[58]
m35 = fixed_investment/df['Opening'].iloc[39]
m36 = fixed_investment/df['Opening'].iloc[19]

In [20]:
sh3 = m25+m26+m27+m28+m29+m30+m31+m32+m33+m34+m35+m36
print(f"You owned a total of {sh3} shares")
v3 = sh3*df['Closing Price'].iloc[19]
print(f"The value of your shares : {v3} riels")
print("You invested", fixed_investment*12, "riels")

rd3 = round(v2/(fixed_investment*12) - 1, 4)
print(f"The return for year 3 : {rd3} returns")

You owned a total of 1115.1448262261117 shares
The value of your shares : 12690348.122453151 riels
You invested 12000000 riels
The return for year 3 : 0.0151 returns


# **Anuual Returns of Both Strategies**

In [21]:
lump = [r1, r2, r3]
dca = [rd1, rd2, rd3]

print('Average of Annual Returns of LS: ', round(np.mean(lump), 4))
print('Average of Annual Returns of DCA: ', round(np.mean(dca), 4))
print('Standard Deviations of Annual Returns of LS:', round(np.std(lump), 4))
print('Standard Deviations of Annual Returns of DCA:', round(np.std(dca), 4))

Average of Annual Returns of LS:  -0.1317
Average of Annual Returns of DCA:  -0.0218
Standard Deviations of Annual Returns of LS: 0.133
Standard Deviations of Annual Returns of DCA: 0.0522


# **Test for Two Population Means**

### **Method 1 : Using Critical Region**

Test $H_0$ : $\mu_1 - \mu_2 = 0$  vs  $H_a$ : $\mu_1 - \mu_2 \ne 0$

We have 

significance level $\alpha = 0.05$

$\bar{X} = -0.1317$

$\bar{Y} = -0.0435$ 

$m=n=735$

$s_1 = 0.133$

$s_2 = 0.0828$


Test statistic Value $ z = \dfrac{\bar{X} - \bar{Y} - (\mu_1 - \mu_2)}{\sqrt{\dfrac{s_1^2}{m} + \dfrac{s_2^2}{n}}} $
$ = \dfrac{-0.1317+0.0435}{\sqrt{\dfrac{0.133^2}{735} + \dfrac{0.0828^2}{735}}} = -15.26$


$z_{\frac{\alpha}{2}} = \phi^{-1}(1 - \frac{\alpha}{2}) = \phi^{-1}(0.975) = 1.96$ 

$\text{Critical Region } C = \{ z: |z| \geq z_{\frac{\alpha}{2}} = 1.96 \}$ 

Since $z \in C$, we decide to reject $H_0$ at $\alpha = 0.05$.

Thus, there is enough evidence to support the claim $\mu_1 \ne \mu_2$ and we can say there are significant differences in the average annual returns between Investment Strategy A and Investment Strategy B.

### **Method 2 : Using Confidence Interval**

$CI(\mu_1 - \mu_2) = \left[ \bar{X} - \bar{Y} \pm z_{\frac{\alpha}{2}} \sqrt{\dfrac{s_1^2}{m} + \dfrac{s_2^2}{n}} \right] = \left[-0.11, -0.065\right]$

Since $0 \notin CI$,

Therefore, we decide to reject the null hypothesis that $\mu_1 - \mu_2 = 0$

# **Test for 2 Population Varainces**

### **Method 1 : Using Critical Region**

Test $H_0$ : $\sigma_1^2 = \sigma_2^2$ vs $H_a$ : $\sigma_1^2 \ne \sigma_2^2$

We have

significance level $\alpha = 0.05$

$s_1 = 0.133$

$s_2 = 0.0828$

$m=n=735$

Test statistic value $f = \dfrac{s_1^2}{s_2^2} = \dfrac{0.133^2}{0.0828^2} = 2.58$

Critical Region $C = \{ f : f \geq F_{\frac{\alpha}{2} , m-1, n-1} \text{ or } f \leq F_{1- \frac{\alpha}{2} , m-1, n-1}  \}$

$F_{\frac{\alpha}{2} , m-1, n-1} = F_{0.025, 734, 734} = 1.15 $

$F_{1- \frac{\alpha}{2} , m-1, n-1} = F_{0.975, 734, 734} = 0.86$

Since $f \in C$, then $H_0$ is rejected.

Thus, we could say the distribution of risk (standard deviation of returns) vary significantly between the two investment strategies.

### **Method 2 : Using Confidence Interval**

$CI\left(\dfrac{\sigma_1}{\sigma_2}\right) = \left[ \dfrac{\sigma_1}{\sigma_2} \cdot \dfrac{1}{\sqrt{F_{\frac{\alpha}{2} , m-1, n-1} }},
\dfrac{\sigma_1}{\sigma_2} \cdot \sqrt{F_{\frac{\alpha}{2} , m-1, n-1} }\right] = [1.49, 1.72]$

Since $1 \notin CI$,

Therefore, we decide to reject the Null Hypothesis that $\sigma_1^2 = \sigma_2^2$